### ISSUE TITLE
Possible infinite loop in FrameLoop.ts `update` method

### ISSUE THREAD

**Issue by thatsprettyfaroutman** [react-spring/react-spring#770]
## 🐛 Bug Report

Possible infinite loop in `FrameLoop.ts` update method on low end machines 📠

In this part of the code (line 74): 

```
if (time > lastTime + 64) lastTime = time; // http://gafferongames.com/game-physics/fix-your-timestep/

let numSteps = Math.floor(time - lastTime);
```

What happens is when the delta time between two frames is longer than 64 ms `numSteps` will become `0` and the animation won't move forward, causing it to get stuck

## To Reproduce

Start a low end machine on vm where delta time between frames gets longer than 64 ms. Alternatively force the `numSteps` to always be zero.

## Expected behavior

Animations do not get stuck on low frame rate

## Link to repro (highly encouraged)

https://codesandbox.io/s/fervent-carson-n4m3p

## Environment

- `react-spring` v8.0.23
- `react` v16.8.6

## Fix

Remove the line 74: `if (time > lastTime + 64) lastTime = time;`

**Comment by thatsprettyfaroutman** [comment]
Possibly related to #767 and #720

**Comment by brunolemos** [comment]
I think this happens on high end machines some times as well.

I am investigating performance issues on my app (on a macpro) and there is something wrong going on. You can see that the FrameLoop is calling `update` a lot of times event though apparently all animations have already finished:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/619186/62417619-5dcb5880-b62a-11e9-8215-ce09479a812a.png">

This causes the animation to be super laggy.
(on both v8 and v9-31)

**Comment by brunolemos** [comment]
I think it has something to do with having lot's of Animated components rendered in the screen. If I render only one the issue doesn't happen, but if I render more things it happens. And that would explain #767 as well.

**Comment by brunolemos** [comment]
I made some investigations.

1. It seems to be related to `requestAnimationFrame`. Overriding this with `window.requestIdleCallback` fixes things on development (but not on production, for some reason).

    ```tsx
    import { Globals } from 'react-spring'

    Globals.assign({`
      requestAnimationFrame: (window as any).requestIdleCallback,
      cancelAnimationFrame: (window as any).cancelIdleCallback,
    })
    ```

2. When playing with different `requestAnimationFrame` overrides, I started getting this error: `InternalError: Too much recursion`. I checked the line and saw that react-spring searches for Animated components in the whole subtree. Dunno if that's related to this issue, but may be one of the performance bottlenecks.

    <img width="484" alt="image" src="https://user-images.githubusercontent.com/619186/62418015-bc4a0400-b635-11e9-8a90-69f3c71e9177.png">

3. Here's a screenshot of the performance tab, notice that the painting is expense for every frame:
    <img width="615" alt="image" src="https://user-images.githubusercontent.com/619186/62418402-8b220180-b63e-11e9-82d8-5f8b285798b3.png">


    Mozilla docs page has some warnings related to `style flush` and `requestAnimationFrame` that I think are related to what's happening. The whole article is worth reading: https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Performance_best_practices_for_Firefox_fe_engineers#Detecting_and_avoiding_synchronous_style_flushes

    <img width="687" alt="image" src="https://user-images.githubusercontent.com/619186/62418376-fc14e980-b63d-11e9-97d8-57e8a59c91d9.png">


    <img width="696" alt="image" src="https://user-images.githubusercontent.com/619186/62418195-8ce9c600-b63a-11e9-9bfd-1f0fa536fe03.png">

4. There's this [list](https://gist.github.com/paulirish/5d52fb081b3570c81e3a) specifing what forces a layout paint / reflow, it's funny that it says `for loops that force layout & change the DOM are the worst, avoid them`, I think that's exactly what react-spring does but I still don't know what exactly could be done to improve this.

**Comment by brunolemos** [comment]
I don't know if the problem is an infinite loop, it seems to be more a "each step of the loop is too slow" so the loop takes a looooong time.

**Comment by thatsprettyfaroutman** [comment]
Ah, I see! I wonder if the jank is needed tho? At least in my case it seems to be causing more problems than it solves.

You see, we use react-spring in our product and we support low end ie11 machines, where the problem gets very severe. I have a case where a transition animation between views always takes more than 200ms per frame. This causes the transition animation to always be janked on every frame and the animation never to proceeds and won't be able to finish. Causing it to get stuck in the animation loop.

However if the jank is disabled these machines are still able to at least finish the animation. Frame rate is low, like 5 fps or something, but the animations finish and the app is usable.

**Comment by thatsprettyfaroutman** [comment]
Yeah, you're right its not really an infinite loop, but never ending animation loop or something like that

**Comment by phaistonian** [comment]
Is it me or `Globals.assign` is gone in the latest beta?

**Comment by aleclarson** [comment]
Fixed in #808
